src/sage/matroids/chow_ring.py

```diff
@@ -0,0 +1,349 @@
+"""
+
+from sage.matroids.chow_ring_ideal import ChowRingIdeal_nonaug, AugmentedChowRingIdeal_fy, AugmentedChowRingIdeal_atom_free
+from sage.rings.quotient_ring import QuotientRing_generic
+from sage.categories.graded_algebras_with_basis import GradedAlgebrasWithBasis
+from sage.categories.commutative_rings import CommutativeRings
+from sage.combinat.posets.posets import Poset
+from itertools import product, combinations
+
+class ChowRing(QuotientRing_generic):
+    r"""
+    The Chow ring of a matroid.
+
+    The *Chow ring of a matroid* `M` is defined as the quotient ring:
+
+    .. MATH::
+
+        A^*(M)_R := R[x_{F_1}, \ldots, x_{F_k}] / (Q_M + L_M)
+
+    where `(Q_M + L_M)` is the Chow ring ideal of matroid `M`.
+
+    The *augmented Chow ring of matroid* `M` in the Feitchner-Yuzvinsky presentation
+    is the quotient ring:
+
+    ..MATH::
+
+        A(M)_R := R[y_{e_1}, \ldots, y_{e_n}, x_{F_1}, \ldots, x_{F_k}] / (I_M + J_M)
+
+    where `(I_M + J_M)` is the augmented Chow ring ideal of matroid `M`
+    of Feitchner-Yuzvinsky presentation.
+
+    The *augmented Chow ring of atom-free presentation* is the quotient ring:
+
+    ..MATH::
+
+        A(M)_R := R[x_{F_1}, \ldots, x_{F_k}] / I_{af}M
+
+    where `I_{af}M` is the augmented Chow ring ideal of matroid `M` of atom-free presentation.
+
+    .. SEEALSO::
+
+        :mod:`sage.matroids.chow_ring_ideal`
+    
+    INPUT:
+
+    - `M` -- a matroid
+    - `R` -- a commutative ring
+    - ``augmented`` -- boolean; when ``True``, this is the augmented
+        Chow ring and if ``False``, this is the non-augmented Chow ring
+    - ``presentation`` -- string (default: ``None``); one of the following:
+    
+      * ``"fy"`` - Feitchner-Yuzvinsky presentation*
+      * ``"atom-free" - Atom-free presentation*
+
+
+    REFERENCES:
+
+    - [FY2004]_
+    - [AHK2015]_
+
+    EXAMPLES::
+
+        sage: M1 = matroids.catalog.P8pp() #more examples
+        sage: ch = M1.chow_ring(QQ, False)
+        sage: ch
+        Chow ring of P8'': Matroid of rank 4 on 8 elements with 8 nonspanning circuits
+    """
+    def __init__(self, R, M, augmented, presentation=None):
+        r"""
+        Initialize ``self``.
+
+        EXAMPLES::
+
+            sage: ch = matroids.Wheel(3).chow_ring(QQ, False)
+            sage: TestSuite(ch).run()
+        """
+        self._matroid = M
+        self._augmented = augmented
+        self._presentation = presentation
+        if augmented is True:
+            if presentation == 'fy':
+                self._ideal = AugmentedChowRingIdeal_fy(M, R)
+            elif presentation == 'atom-free':
+                self._ideal = AugmentedChowRingIdeal_atom_free(M, R)
+        else:
+            self._ideal = ChowRingIdeal_nonaug(M, R)
+        C = CommutativeRings().Quotients() & GradedAlgebrasWithBasis(R).FiniteDimensional()
+        QuotientRing_generic.__init__(self, R=self._ideal.ring(), I=self._ideal, names=self._ideal.ring().variable_names(), category=C)
```

Comment:
```suggestion
        QuotientRing_generic.__init__(self, R=self._ideal.ring(),
                                      I=self._ideal,
                                      names=self._ideal.ring().variable_names(),
                                      category=C)
```
So it is closer to 80 char/ line.